CRM/Utils/Token.php

```diff
@@ -657,17 +657,14 @@ public static function &replaceContactTokens(
     $returnBlankToken = FALSE,
     $escapeSmarty = FALSE
   ) {
-    $key = 'contact';
-    if (self::$_tokens[$key] == NULL) {
-      // This should come from UF
-
-      self::$_tokens[$key]
-        = array_merge(
-          array_keys(CRM_Contact_BAO_Contact::exportableFields('All')),
-          array('checksum', 'contact_id')
-        );
-    }
+    // Refresh contact tokens in case they have changed. There is heavy caching
+    // in exportable fields so there is no benefit in doing this conditionally.
+    self::$_tokens['contact'] = array_merge(
```

Comment:
ditch static var which makes tests hard & does not acheive much

Reply:
:+1: